BasicSample/Assets/Editor/MixedRealitySampleQuickSetupWindow.cs

```diff
@@ -0,0 +1,237 @@
+using UnityEditor.XR.OpenXR.Features;
+using UnityEngine;
+using UnityEngine.XR.Management;
+using UnityEngine.XR.OpenXR;
+using UnityEngine.XR.OpenXR.Features;
+
+
+namespace Microsoft.MixedReality.OpenXR.BasicSample
+{
+    enum MixedRealityProjectConfiguration
+    {
+        None,
+        RunNativelyOnPCVR,
+        RunNativelyOnHL2,
+        RunRemotelyOnUWP,
+        RunRemotelyOnWin32
+    }
+    public class MixedRealitySampleQuickSetupWindow : EditorWindow
+    {
+        private MixedRealityProjectConfiguration m_selectedMRConfiguration;
+        private bool m_disablePopup;
+        private const float Default_Window_Height = 700.0f;
+        private const float Default_Window_Width = 500.0f;
+        private const float Default_Label_Width = 400f;
+
+        public static MixedRealitySampleQuickSetupWindow Instance { get; private set; }
+        public static bool IsOpen => Instance != null;
+        private static PopupUserSettings UserSettings;
+        private const string SettingsFileName = "MixedRealitySampleQuickSetupSettings.asset";
+        private static string UserSettingsFolder => Path.Combine(Application.dataPath, "..", "UserSettings");
+        private static string SettingsAssetPath => Path.Combine(UserSettingsFolder, SettingsFileName);
+
+        [MenuItem("Mixed Reality/Quick Setup", false, 499)]
+        private static void ShowWindowFromMenu()
+        {
+            ShowWindow(true);
+        }
+
+        public static void ShowWindow(bool showfromMenu)
+        {
+            GetUserSettings();
+            // There should be only one quick setup window open as a "pop-up". If already open, then just force focus on our instance
+            if (IsOpen)
+            {
+                Instance.Focus();
+            }
+            else
+            {
+                if(!UserSettings.DisablePopup || showfromMenu)
+                {
+                    var window = CreateInstance<MixedRealitySampleQuickSetupWindow>();
+                    window.titleContent = new GUIContent("MixedReality Sample Quick Setup", EditorGUIUtility.IconContent("_Popup").image);
+                    window.position = new Rect(Screen.width / 2.0f, Screen.height / 2.0f, Default_Window_Width, Default_Window_Height);
+                    window.ShowUtility();
+                }
+
+            }
+        }
+
+        internal static void GetUserSettings()
+        {
+            if(UserSettings == null)
+            {
+                UserSettings = CreateInstance<PopupUserSettings>();
+
+                if (File.Exists(SettingsAssetPath))
+                {
+                    using (StreamReader settingsReader = new StreamReader(SettingsAssetPath))
+                    {
+                        JsonUtility.FromJsonOverwrite(settingsReader.ReadToEnd(), UserSettings);
+                    }
+                }
+                else
+                {
+#pragma warning disable CS0618 // to use the obsolete fields to port to the new asset file
```

Comment:
I don't think you need these `#pragma` lines, since you aren't using any obsolete fields here as far as I can tell

Reply:
You may not even need this `else` case generally, since `DisablePopup` looks like it initializes to `false` anyway

Reply:
addressed